client/src/main.rs

```diff
@@ -12,205 +12,271 @@
 use tonic::transport::Channel;
 use tonic::Request;
 
-/** Public Types **/
-pub type Result<T> = std::result::Result<T, Box<dyn Error>>;
+use crate::helper::ClientHelper;
+
+/// Default host to connect to.
+const DEFAULT_HOST: &str = "127.0.0.1";
+
+/// Default port to connect to. The server and client currently does not support changing the port.
+const DEFAULT_PORT: u16 = 9999;
 
-/** Public Methods **/
-fn main() {
+/// Error to emit when the server does not provide a response when one is expected.
+const TRANSPORT_ERROR: &str = "transport error: no messages received.";
+
+/// Parse the command line arguments to extract the host running the server to connect to and the
+/// file containing the queries to execute on the server. If the server host is not provided it
+/// defaults to [`DEFAULT_HOST`], and if the file containing queries is not provided a
+/// read-eval-print loop is opened. Returns [`String`] if the command line arguments cannot be
+/// parsed, the client cannot connect to the server, or the file containing the queries cannot be
+/// read.
+fn main() -> Result<(), String> {
+    // Parse the command line arguments.
     let args = env::args();
     if args.len() > 3 {
-        // The errors are consciously ignored as the program is terminating.
+        // The errors are consciously ignored as the client is terminating.
         let binary_path = env::current_exe().unwrap();
         let binary_name = binary_path.file_name().unwrap();
-        println!(
+        Err(format!(
             "usage: {} [server_address] [query_file]",
             binary_name.to_str().unwrap()
-        );
-        return;
+        ))?;
     }
-    let (address, query_file) = parse_arguments(args);
-
-    if let Ok(rt) = tokio::runtime::Runtime::new() {
-        let address = address.unwrap_or_else(|| "127.0.0.1".to_string());
-        match connect(&rt, &address, 9999) {
-            Ok(fsc) => {
-                // Execute queries.
-                let result = if let Some(query_file) = query_file {
-                    file(rt, fsc, &query_file)
-                } else {
-                    repl(rt, fsc)
-                };
-
-                // Print error.
-                match result {
-                    Ok(_) => (),
-                    Err(message) => eprintln!("{}", message),
-                };
-            }
-            Err(message) => eprintln!("error: cannot connect to {} due to a {}", address, message),
-        }
+    let (maybe_host, maybe_query_file_path) = parse_command_line_arguments(args);
+
+    // Create the Tokio runtime.
+    let runtime = Runtime::new()
+        .map_err(|error| format!("error: unable to initialize run-time due to {}", error))?;
+
+    // Connect to the server.
+    let host = maybe_host.unwrap_or_else(|| DEFAULT_HOST.to_owned());
+    let flight_service_client = connect(&runtime, &host, DEFAULT_PORT).map_err(|error| {
+        format!(
+            "error: cannot connect to {} due to a {}",
```

Comment:
It does not seem like we use the `error:` prefix anywhere in the server. Maybe remove this.

Reply:
Fixed.